### ISSUE TITLE
twisted/mail/test/test_mail.py should use setUp and tearDown instead of setUpClass and tearDownClass

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2746]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2746|
|Type|task|
|Created|2007-07-20 14:43:11Z|

ProcessAliasTestCase clobbers the highly dubious DNSNAME attribute of the smtp module in a setUpClass and restores it in tearDownClass.  There's no reason not to do this in setUp/tearDown instead.

<details><summary>Searchable metadata</summary>

```
trac-id__2746 2746
type__task task
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1184942591000000 1184942591000000
changetime__1185626766000000 1185626766000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Done in setup-teardown-mail-2746

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @exarkun|
|-|-|

Fine, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [20858]) Merge setup-teardown-mail-2746

Author: exarkun
Reviewer: therve
Fixes [#2746](https://github.com/twisted/twisted/issues/2746)

Replace setUpClass and tearDownClass on ProcessAliasTestCase, which mocked and
then replaced the original value of smtp.DNSNAME, which setUp and tearDown
methods which do the same thing.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
